samples/ConPTY/EchoCon/EchoCon/EchoCon.cpp

```diff
@@ -91,52 +78,50 @@ int main()
                     free(startupInfo.lpAttributeList);
                 }
             }
-            else
-            {
-                wprintf_s(L"Error: Failed to initialize StartupInfo attached to ConPTY [0x%x]",
-                    GetLastError());
-            }
-        }
-        else
-        {
-            wprintf_s(L"Error: Failed to create pipes");
-        }
 
-        // Clean-up the pipes
-        if (hPipePTYOut != INVALID_HANDLE_VALUE) CloseHandle(hPipePTYOut);
-        if (hPipePTYIn != INVALID_HANDLE_VALUE) CloseHandle(hPipePTYIn);
-        if (hPipeOut != INVALID_HANDLE_VALUE) CloseHandle(hPipeOut);
-        if (hPipeIn != INVALID_HANDLE_VALUE) CloseHandle(hPipeIn);
+            // Close ConPTY - this will terminate client process if running
+            ClosePseudoConsole(hPC);
 
-        // Restore console to its original mode.
-        hr = SetConsoleMode(hConsole, consoleMode)
-            ? S_OK
-            : GetLastError();
+            // Clean-up the pipes
+            if (INVALID_HANDLE_VALUE != hPipeOut) CloseHandle(hPipeOut);
```

Comment:
y u switch these conditionals around?

Reply:
I prefer `<const> == <variable>` to protect against accidental assignment. Ol' trick I learned from Scott Myers' "[Effective C++](https://www.amazon.com/Effective-Specific-Improve-Programs-Designs/)" that's saved my butt on several occasions.